adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -379,6 +390,103 @@ func CreateLuaCluster(interceptorCerts map[string][]byte, endpoint model.Interce
 	return processEndpoints(endpoint.ClusterName, &endpoint.EndpointCluster, interceptorCerts, endpoint.ClusterTimeout, endpoint.EndpointCluster.Endpoints[0].Basepath)
 }
 
+// CreateRateLimitCluster creates cluster relevant to the rate limit service
+func CreateRateLimitCluster() (*clusterv3.Cluster, []*corev3.Address, error) {
+	conf, _ := config.ReadConfigs()
+	var sslCertSanHostName string = ""
+	if conf.Envoy.RateLimit.SSLCertSANHostname == "" {
+		sslCertSanHostName = conf.Envoy.RateLimit.Host
+	} else {
+		sslCertSanHostName = conf.Envoy.RateLimit.SSLCertSANHostname
+	}
+	rlCluster := &model.EndpointCluster{
+		Endpoints: []model.Endpoint{
+			{
+				Host:    conf.Envoy.RateLimit.Host,
+				URLType: httpsURLType,
+				Port:    conf.Envoy.RateLimit.Port,
+			},
+		},
+	}
+	cluster, address, rlErr := processEndpoints(rateLimitClusterName, rlCluster, nil, 20, "")
```

Comment:
Shall we log the error and return the error if not nil from here?

Reply:
It's happening in here.
https://github.com/wso2/product-microgateway/blob/1f95eb5794419ae5d9b8cb63ef0cf6ecfd89bb0a/adapter/internal/oasparser/config_generator.go#L64

Reply:
fixed with: https://github.com/wso2/product-microgateway/pull/3142/commits/2b7f96496d9660a5e4c51b7531d65c17785dd869